### PR TITLE
Spring 07

### PR DESCRIPTION
**Pull request by Alex-Korzun**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 1]
```diff
@@ -34,15 +36,19 @@ public OrderController(OrderService orderService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto complete(@RequestParam Long id) {
-        return orderMapper
-                .toDto(orderService.completeOrder(shoppingCartService
-                        .getByUser(userService.get(id))));
+    public OrderResponseDto complete(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
```
Comment (gigbat) [inline src/main/java/com/dev/cinema/controller/OrderController.java:40]: if you wanna use that, then you can write like: 
```suggestion
        UserDetails userDetails = authentication.getName()
```

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 2]
```diff
@@ -34,15 +36,19 @@ public OrderController(OrderService orderService,
...
+        String email = userDetails.getUsername();
+        User user = userService.findByEmail(email).get();
+        return orderMapper.toDto(orderService.completeOrder(shoppingCartService.getByUser(user)));
     }
 
     @GetMapping
-    public List<OrderResponseDto> getOrdersById(@RequestParam Long id) {
-        return orderService.getOrdersHistory(userService.get(id))
+    public List<OrderResponseDto> getOrdersById(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
```
Comment (gigbat) [inline src/main/java/com/dev/cinema/controller/OrderController.java:48]: same

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -35,15 +37,19 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/movie-sessions")
-    public void add(@RequestParam Long userId, Long movieSessionId) {
-        User user = userService.get(userId);
+    public void add(Authentication authentication, @RequestParam Long movieSessionId) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
```
Comment (gigbat) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java:41]: same

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -35,15 +37,19 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
...
+        User user = userService.findByEmail(email).get();
         MovieSession movieSession = movieSessionService.get(movieSessionId);
         shoppingCartService.addSession(movieSession, user);
     }
 
     @GetMapping("/by-user")
-    public ShoppingCartResponseDto getByUser(@RequestParam Long id) {
-        User user = userService.get(id);
+    public ShoppingCartResponseDto getByUser(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
```
Comment (gigbat) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java:50]: and here

**Review by gigbat — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controller/OrderController.java (hunk 1 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/controller/OrderController.java:40]: I don't know if that is right option. `getName()` returns String. And `getPrincipal()` returns an Object of `UserDetails`. Or you mean get email with `getName()` method without creating `userDetails`?

**Review by gigbat — APPROVED** [review]

File: src/main/java/com/dev/cinema/controller/OrderController.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/controller/OrderController.java:40]: yes, we can get email with getName() method without creating userDetails

File: src/main/java/com/dev/cinema/validation/PasswordValidation.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Constraint(validatedBy = PasswordValidationImpl.class)
+@Target({ ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface PasswordValidation {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/validation/PasswordValidation.java]: redundant empty line

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**